atc/creds/secretsmanager/secretsmanager.go

```diff
@@ -3,69 +3,60 @@ package secretsmanager
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/secretsmanager"
 	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
-
-	varTemplate "github.com/cloudfoundry/bosh-cli/director/template"
 )
 
 type SecretsManager struct {
 	log             lager.Logger
 	api             secretsmanageriface.SecretsManagerAPI
-	TeamName        string
-	PipelineName    string
-	SecretTemplates []*template.Template
+	secretTemplates []*template.Template
 }
 
-func NewSecretsManager(log lager.Logger, api secretsmanageriface.SecretsManagerAPI, teamName string, pipelineName string, secretTemplates []*template.Template) *SecretsManager {
+func NewSecretsManager(log lager.Logger, api secretsmanageriface.SecretsManagerAPI, secretTemplates []*template.Template) *SecretsManager {
 	return &SecretsManager{
 		log:             log,
 		api:             api,
-		TeamName:        teamName,
-		PipelineName:    pipelineName,
-		SecretTemplates: secretTemplates,
+		secretTemplates: secretTemplates,
 	}
 }
 
-func (s *SecretsManager) buildSecretId(nameTemplate *template.Template, secret string) (string, error) {
-	var buf bytes.Buffer
-	err := nameTemplate.Execute(&buf, &Secret{
-		Team:     s.TeamName,
-		Pipeline: s.PipelineName,
-		Secret:   secret,
-	})
-	return buf.String(), err
-}
-
-func (s *SecretsManager) Get(varDef varTemplate.VariableDefinition) (interface{}, bool, error) {
-	for _, st := range s.SecretTemplates {
-		secretId, err := s.buildSecretId(st, varDef.Name)
-		if err != nil {
-			s.log.Error("build-secret-id", err, lager.Data{"template": st.Name(), "secret": varDef.Name})
-			return nil, false, err
-		}
+// NewSecretLookupPaths defines how variables will be searched in the underlying secret manager
+func (s *SecretsManager) NewSecretLookupPaths(teamName string, pipelineName string) []creds.SecretLookupPath {
+	lookupPaths := []creds.SecretLookupPath{}
+	for _, tmpl := range s.secretTemplates {
+		lPath := NewSecretLookupPathAws(tmpl, teamName, pipelineName)
 
-		if strings.Contains(secretId, "//") {
-			continue
+		// if pipeline name is empty, double slashes may be present in the rendered template
+		// let's avoid adding these templates
+		samplePath, err := lPath.VariableToSecretPath("variable")
+		if err == nil && !strings.Contains(samplePath, "//") {
+			lookupPaths = append(lookupPaths, lPath)
 		}
+	}
+	return lookupPaths
+}
 
-		value, found, err := s.getSecretById(secretId)
-		if err != nil {
-			s.log.Error("get-secret", err, lager.Data{
-				"template": st.Name(), "secret": varDef.Name, "secretId": secretId,
-			})
-			return nil, false, err
-		}
-		if found {
-			return value, true, nil
-		}
+// Get retrieves the value and expiration of an individual secret
+func (s *SecretsManager) Get(secretPath string) (interface{}, *time.Time, bool, error) {
+	value, expiration, found, err := s.getSecretById(secretPath)
+	if err != nil {
+		s.log.Error("unable to retrieve aws secret", err, lager.Data{
```

Comment:
this should be conventional formatting:

```suggestion
		s.log.Error("failed-to-fetch-aws-secret", err, lager.Data{
```